partiql-cli/src/visualize/ast_to_dot.rs

```diff
@@ -0,0 +1,589 @@
+        self.to_dot(out, &ast.at_alias).edges(out, &id, "at");
+        self.to_dot(out, &ast.by_alias).edges(out, &id, "by");
+
+        vec![id]
+    }
+}
+
+impl ToDot<ast::Join> for AstToDot {
+    fn to_dot(&mut self, out: &mut Scope, ast: &ast::Join) -> Targets {
+        let lbl = match &ast.kind {
+            ast::JoinKind::Inner => "Inner Join",
+            ast::JoinKind::Left => "Left Join",
+            ast::JoinKind::Right => "Right Join",
+            ast::JoinKind::Full => "Full Join",
+            ast::JoinKind::Cross => "Cross Join",
+        };
+        let id = out.node_auto_labelled(lbl).id();
+
+        self.to_dot(out, &ast.left).edges(out, &id, "left");
+        self.to_dot(out, &ast.right).edges(out, &id, "right");
+        self.to_dot(out, &ast.predicate)
+            .edges(out, &id, "predicate");
+
+        vec![id]
+    }
+}
+
+impl ToDot<ast::JoinSpec> for AstToDot {
+    fn to_dot(&mut self, out: &mut Scope, ast: &ast::JoinSpec) -> Targets {
+        match &ast {
+            ast::JoinSpec::On(fl) => {
+                let id = out.node_auto_labelled("On").id();
+                self.to_dot(out, fl).edges(out, &id, "");
+                vec![id]
+            }
+            ast::JoinSpec::Using(j) => {
+                let id = out.node_auto_labelled("Using").id();
+                self.to_dot(out, j).edges(out, &id, "");
+                vec![id]
+            }
+            ast::JoinSpec::Natural => vec![out.node_auto_labelled("Natural").id()],
+        }
+    }
+}
+
+impl ToDot<ast::Call> for AstToDot {
+    fn to_dot(&mut self, out: &mut Scope, ast: &ast::Call) -> Targets {
+        let id = out.node_auto_labelled("Call").id();
+
+        self.to_dot(out, &ast.func_name).edges(out, &id, "name");
+        self.to_dot(out, &ast.args).edges(out, &id, "args");
+
+        vec![id]
+    }
+}
+
+impl ToDot<ast::CallArg> for AstToDot {
+    fn to_dot(&mut self, out: &mut Scope, ast: &ast::CallArg) -> Targets {
+        match ast {
+            ast::CallArg::Star() => vec![out.node_auto_labelled("*").id()],
+            ast::CallArg::Positional(e) => self.to_dot(out, e),
+            ast::CallArg::Named { name, value } => {
+                let id = out.node_auto_labelled("Named").id();
+                self.to_dot(out, name).edges(out, &id, "name");
+                self.to_dot(out, value).edges(out, &id, "value");
+                vec![id]
+            }
+        }
+    }
+}
+
+impl ToDot<ast::CallAgg> for AstToDot {
+    fn to_dot(&mut self, out: &mut Scope, ast: &ast::CallAgg) -> Targets {
+        let lbl = match &ast.setq {
+            Some(ast::SetQuantifier::Distinct) => "CallAgg | Distinct",
+            _ => "CallAgg | All",
+        };
+        let id = out.node_auto_labelled(lbl).id();
+
+        self.to_dot(out, &ast.func_name).edges(out, &id, "name");
+        self.to_dot(out, &ast.args).edges(out, &id, "args");
+
+        vec![id]
+    }
+}
+
+impl ToDot<ast::Path> for AstToDot {
```

Comment:
I played around a bit with some of the conformance tests and noticed the ast for some of the expressions may not be correct. Using the "[complicated nested path](https://github.com/partiql/partiql-tests/blob/ef3d21be1250423d321299bb276455a10b187a21/partiql-tests-data/success/syntax/primitives/path-expression.ion#L57-L65)" test from `partiql-tests`, I got the following result:

```
❯ cargo run --release --features visualize -- ast -Tpng '''a.b.c['item']."d"[5].e['s'].f[1+2]''' > /tmp/ast.png
```
![ast](https://user-images.githubusercontent.com/23245405/180897057-05a17116-5a82-4e9d-b22a-4ca43f1c7eea.png)

The parsed AST output by `parse_partiql` doesn't include nodes along the path other than the start and final nodes (for this example test, only has `a`, and `f[1+2]`).

Reply:
Seems to be a bug; cut #159 for looking into this.